### ISSUE TITLE
Switch to using a cryptographically secure random number generator

### ISSUE THREAD

**Issue by diafygi** [markbao/speakeasy#40]
`Math.random()` is used to generate the initial key[1]. Please switch to using `crypto.randomBytes()`.

[1] - https://github.com/markbao/speakeasy/blob/d9525fdde341624109557da52ad6cdd270025059/lib/speakeasy.js#L254

**Comment by coolaj86** [comment]
```
//
// Generate a 20-byte (160-bit) random OTP key as base32 ascii
//
function generateGoogleAuthKey() {
  // 20 cryptographically random binary bytes (160-bit key)
  var key = require('crypto').randomBytes(20);

  // 32 ascii characters without trailing '='s
  var base32 = require('thirty-two').encode(key).toString('utf8').replace(/=/g, '');

  // lowercase with a space every 4 characters
  var formatted = base32.toLowerCase().replace(/(\w{4})/g, "$1 ").trim();

  return formatted;
}
```

**Comment by coolaj86** [comment]
[Node Authenticator](https://github.com/Daplie/node-authenticator)
================

I wrote [Node Authenticator](https://github.com/Daplie/node-authenticator) as a replacement for SpeakEasy. It uses node's native [crypto](https://nodejs.org/api/crypto.html) module and [notp](https://github.com/guyht/notp).

* https://github.com/Daplie/node-authenticator

**Comment by markbao** [comment]
Thanks @diafygi. Closed thanks to #33.
